src/System.Management.Automation/engine/MshMemberInfo.cs

```diff
@@ -1909,9 +1909,18 @@ internal class PSMethodInvocationConstraints
         internal PSMethodInvocationConstraints(
             Type methodTargetType,
             Type[] parameterTypes)
+            : this(methodTargetType, genericTypeParameters: null, parameterTypes)
         {
-            this.MethodTargetType = methodTargetType;
+        }
+
+        internal PSMethodInvocationConstraints(
+            Type methodTargetType,
+            Type[] genericTypeParameters,
+            Type[] parameterTypes)
+        {
+            MethodTargetType = methodTargetType;
             _parameterTypes = parameterTypes;
+            GenericTypeParameters = genericTypeParameters ?? Array.Empty<Type>();
```

Comment:
Maybe this should just be `GenericTypeParameters = genericTypeParameters;`, just like how `_parameterTypes` is handled.

Reply:
Addressed